src/simulators/statevector/statevector_state.hpp

```diff
@@ -556,14 +556,13 @@ void State<statevec_t>::set_config(const Config &config) {
   }
 
   // Set OMP threshold for state update functions
-  omp_qubit_threshold_ = config.statevector_parallel_threshold;
+  config.get_option(omp_qubit_threshold_, "statevector_parallel_threshold");
 
   // Set the sample measure indexing size
-  if (config.statevector_sample_measure_opt) {
-    int index_size = config.statevector_sample_measure_opt;
-    for (int_t i = 0; i < BaseState::qregs_.size(); i++) {
+  int index_size;
+  if (config.get_option(index_size, "statevector_sample_measure_opt")) {
+    for (int_t i = 0; i < BaseState::qregs_.size(); i++)
       BaseState::qregs_[i].set_sample_measure_index_size(index_size);
-    }
```

Comment:
This still doesn't look exactly correct to me (or maybe it's a different instance of a bug I commented on elsewhere).  `config.get_option` returns a `bool` that indicates whether the option exists, not the _value_ of the option, which is stored into the `index_size` reference - if `index_size` is 0, then in the old code the `if` wouldn't be entered, whereas here it will be.

Reply:
Thanks... You are right. Checking value of `index_size` is necessary here.